config/initializers/20_paperclip.rb

```diff
@@ -1,15 +1,28 @@
-if AWSConfig.config.aws_access_id.blank? && AWSConfig.config.aws_secret_key.blank?
-  cred = {
-    bucket: AWSConfig.config.image_bucket_name
-  }
-else
-  cred = {
-    bucket: AWSConfig.config.image_bucket_name,
-    access_key_id: AWSConfig.config.aws_access_id,
-    secret_access_key: AWSConfig.config.aws_secret_key,
+if CloudConfig::service_name == 'aws'
+  if AWSEngine.config.aws_access_id.blank? && AWSEngine.config.aws_secret_key.blank?
+    cred = {
+      bucket: AWSEngine.config.image_bucket_name
+    }
+  else
+    cred = {
+      bucket: AWSEngine.config.image_bucket_name,
+      access_key_id: AWSEngine.config.aws_access_id,
+      secret_access_key: AWSEngine.config.aws_secret_key,
+    }
+  end
+  Paperclip::Attachment.default_options[:storage] = :s3
+  Paperclip::Attachment.default_options[:s3_host_name] = AWSEngine.s3_host_name
+  Paperclip::Attachment.default_options[:s3_region] = AWSEngine.config.region
+  Paperclip::Attachment.default_options[:s3_credentials] = cred
+end
+
+if CloudConfig::service_name == 'azure'
```

Comment:
__[rubocop]__
* Style/ColonMethodCall: Do not use `::` for method calls. [:ref](https://github.com/bbatsov/ruby-style-guide#double-colons)